File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3-coder-30b-mlx_nothink/20260830_231336/go/002_word_frequency/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strings"
	"unicode"
)

func main() {
	file, err := os.Open("input/text.txt")
	if err != nil {
		return
	}
	defer file.Close()

	wordCount := make(map[string]int)

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		words := strings.FieldsFunc(line, func(r rune) bool {
			return !unicode.IsLetter(r)
		})

		for _, word := range words {
			word = strings.ToLower(word)
			if word != "" {
				wordCount[word]++
			}
		}
	}

	if err := scanner.Err(); err != nil {
		return
	}

	type wordFreq struct {
		word  string
		count int
	}

	var sortedWords []wordFreq
	for word, count := range wordCount {
		sortedWords = append(sortedWords, wordFreq{word, count})
	}

	sort.Slice(sortedWords, func(i, j int) bool {
		if sortedWords[i].count == sortedWords[j].count {
			return sortedWords[i].word < sortedWords[j].word
		}
		return sortedWords[i].count > sortedWords[j].count
	})

	for _, wf := range sortedWords {
		fmt.Printf("%s: %d\n", wf.word, wf.count)
	}
}

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3-coder-30b-mlx_nothink/20260830_231336/typescript/002_word_frequency/solution.ts
import fs from 'fs';
import path from 'path';

// Read the input file
const filePath = path.join('input', 'text.txt');
const content = fs.readFileSync(filePath, 'utf8');

// Process the content
const words = content
  .toLowerCase()
  .replace(/[^a-z\s]/g, ' ') // Replace all non-letter characters with spaces
  .split(/\s+/) // Split by whitespace
  .filter(word => word.length > 0); // Remove empty strings

// Count word frequencies
const wordCount: { [key: string]: number } = {};
for (const word of words) {
  wordCount[word] = (wordCount[word] || 0) + 1;
}

// Sort by count descending, then alphabetically ascending
const sortedEntries = Object.entries(wordCount).sort(([wordA, countA], [wordB, countB]) => {
  if (countB !== countA) {
    return countB - countA; // Sort by count descending
  }
  return wordA.localeCompare(wordB); // Sort alphabetically ascending
});

// Output results
for (const [word, count] of sortedEntries) {
  console.log(`${word}: ${count}`);
}